### DISCUSSION TITLE
PCL Emitter

### DISCUSSION THREAD

**Discussion by daverch** [eclipse-birt/birt#1062] [Ideas]
Many years ago my team created a PCL emitter for BIRT. I would like to give to the BIRT project. How would I do that?

**Comment by hvbtup** [comment]
AFAIK the best way is to sign the Eclipse Contributor Assignment, fork the repo, create a branch for your changes and create a pull request.
This is what's technically necessary. Also make sure that it's legal to donate the code, if it was developed by a team.

**Comment by daverch** [comment]
I signed the Eclipse Contributor Assignment, but I don't know how to fork the repo, or create a branch for these changes, or create a pull request. This means I will need a bit of help to give the BIRT project this PCL emitter.

**Comment by hvbtup** [comment]
@wimjongman What would you recommend?

**Comment by ruspl-afed** [comment]
@daverch We have https://github.com/eclipse/birt/blob/master/CONTRIBUTING.md page that describes contribution process.
If instructions are not sufficient there - let's improve it.

For general Git/GitHub user experience like "how to fork the repo" I would recommend to use usual search technique. For example, this is what can be found: https://docs.github.com/en/get-started/quickstart/fork-a-repo

**Comment by wimjongman** [comment]
> I signed the Eclipse Contributor Assignment, but I don't know how to fork the repo, or create a branch for these changes, or create a pull request. This means I will need a bit of help to give the BIRT project this PCL emitter.

@daverch, thanks for the offer, but we can only accept this contribution if you are going to maintain it forever. 

If you can't commit to that, I advise you to package this as a plugin and make it available through Eclipse Marketplace. In this way, people are able to find it easily and maintenance is in your hands.
